src/librustc/Cargo.toml

```diff
@@ -13,12 +13,12 @@ arena = { path = "../libarena" }
 fmt_macros = { path = "../libfmt_macros" }
 graphviz = { path = "../libgraphviz" }
 log = "0.3"
+owning_ref = "0.3.3"
```

Comment:
The reason I mentioned it, fwiw, was that we might have more uses for it.
OTOH, my initial suggestion for this PR was `Box<Deref<Target=[u8]>>`, so `owning_ref` is *potentially* overkill (although I believe it is more efficient).